lib/grape/validations/types/custom_type_coercer.rb

```diff
@@ -137,7 +137,7 @@ def infer_type_check(type)
             # passed, or if the type also implements a parse() method.
             type
           elsif type.is_a?(Enumerable)
-            ->(value) { value.all? { |item| item.is_a? type[0] } }
+            ->(value) { value.respond_to?(:all?) && value.all? { |item| item.is_a? type[0] } }
```

Comment:
`Enumerable` by definition has `all?`, how are we hitting a path where `respond_to?(:all?)` is needed?

Reply:
As I'm now checking for `valid_type?` before `value` is coerced, it means that `value` now could be anything.  The `Proc` that's defined here is used in `def success?` on line 96, which has the annotation of `# This method is called from somewhere within +Virtus::Attribute::value_coerced?+ in order to assert that the value has been coerced successfully.`  `valid_type?` directly calls `value_coerced?`, bringing this full circle.  I think the confusion is around `elsif type.is_a?(Enumerable)`, where `type` refers to the target/desired post-coercion type (eg. `Array`) rather than the type of `value`.

Without it, specs that go from `'1 2 3'` to `[1, 2, 3]` would fail as the type check on `'1 2 3'` would attempt to directly call `'1 2 3'.all? ...` which can't be done on a `String`.

Reply:
Makes sense.